1-Draft/RFCnnnn-Alias-Policy.md

```diff
@@ -0,0 +1,108 @@
+
+    It is even more unpredictable and dangerous to create scripts in the assumption that there are some predefined (standard) aliases because they can be changed by the user or script at any time.
+
+    Modern versions of ScriptAnalyzer already recommend replacing aliases in scripts with their full names.
+
+    >The definition of aliases in modules or scripts and their automatic import becomes an undesirable and dangerous practice.
+
+1. UX in interactive sessions.
+
+    Having many cmdlets and parameters with long names can make it difficult to enter them in interactive sessions.
+    Users who typically use standard settings are typically satisfied with the ```IntellySense``` capabilities.
+    On the other hand, some users actively using interactive sessions may still be annoyed by the need to type names.
+    These users should be able to upload their familiar and useful aliases sets to their interactive sessions.
+    Given the limited functionality of aliases, these users will also configure their environment by executing commands by importing their own functions and various modules using PowerShell profiles.
+    It should be noted that we are planning to add a domain-specific language (DSL). This will provide users with much greater customization capabilities than aliases.
+
+# PowerShell alias policy
+
+1. PowerShell implements standard aliases in all sessions including interactive:
+
+    * % -> ForEach-Object
+    * foreach -> ForEach-Object
+    * ? -> Where-Object
+    * where -> Where-Object
+
+    These aliases are actively used in interactive sessions and scripts.
+    ScriptAnalyzer will continue to recommend not to use them in scripts and modules.
+
+1. By default, PowerShell does not load aliases from modules and InitialSessionState.
+
+    To ensure module backward compatibility:
+    * ```Import-Module -Aliases``` - explicitly loads aliases from the module.
+    * Global configuration option ```GlobalAliasImport``` in ```powershell.config``` turns on loading aliases from modules (and InitialSessionState). It is turned off by default.
+
+1. Backward compatibility of standard aliases.
+
+    To ensure backward compatibility, three sets of aliases are created:
+    * ```Default``` - current alias set (from ```InitialSessionState```) in PowerShell Core. This is the minimum set of aliases that have no conflicts on UNIX systems.
+    * ```Extended``` - the ```Default``` set of aliases and additional popular aliases for Windows users (which can have conflicts on Unix).
+    * ```WindowsPowerShell``` - the full set of standard aliases from Windows PowerShell. This set provides full backward compatibility for Windows users.
+
+    No alias set are loaded by default.
+
+    Global configuration option ```GlobalAliasSet``` in ```powershell.config``` turns on loading the desired set. It is turned off by default.
+
+1. PowerShell prevents the use of aliases in scripts and modules.
```

Comment:
I would like to propose another point to the "alias policy" and that would be to **prevent** the creation of Functions or Aliases which include slashes in the name.  That is:

1. PowerShell prevent modules and users from clobbering "fully qualified" command names.

Reply:
> PowerShell prevent modules and users from clobbering "fully qualified" command names.

We actually use that to mock stuff when unit testing scripts 😛